File: src/fs/exceptions.fs
open System

let DoStuff inError =
    if inError
    then raise (Exception("An error occured!"))

    printfn "Doing stuff!"

try
    DoStuff false
    DoStuff true
    DoStuff false
with
    | _ as ex -> printfn "ERROR: %s" ex.Message

File: src/pl/exceptions.pl
sub DoStuff {
    if ($_[0]){
        die "An error occured!\n";
    }

    print "Doing stuff!\n";
}

eval {
    DoStuff(0);
    DoStuff(1);
    DoStuff(0);
} || do {
    print $@;
}